text/0113-canonical-metadata.md

```diff
@@ -0,0 +1,625 @@
+
+This format was also influenced by standardization choices made with in Salesforce UI platform.
+
+## Entities that will be analyzed
+These are the entities that will be analzed and metadata gathered about:
+* HTML Template file
+    * Custom element references
+        * Tag name
+        * Attributes
+        * Slot content - default and named slots
+    * Static resource
+        * Image urls
+        * Javascript resource
+        * CSS resource
+    * Dynamic components
+    * Template directives
+        * for:each
+        * if
+        * iterator
+        * key
+        * lwc:dom="manual"
+    * Event handlers
+    * Slots
+* Javascript file
+    * Component classes
+        * Inheritance
+        * Public & private properties
+        * Decorators
+        * getters/setters
+        * Public methods
+    * Imports
+        * Builtin modules
+        * External modules
+        * Salesforce scoped modules
+        * Other modules(components and libraries)
+        * Relative imports
+        * Dynamic imports
+    * Exports
+        * Default exports
+        * Named exports
+        * Aggregating modules
+    * JSDoc
+        * Class
+        * Properties
+        * Public methods
+    * Programmatically generated events
+        * type
+        * options: composed, bubbles
+    * Programmatically fetched static resources
+        * `<script>`
+        * `<img>`
+        * `<link>`
+* CSS file
+    * Tokens
+
+## Metadata shape
+
+```json
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "description": "A representation of metadata gathered for a LWC bundle",
+    "type": "object",
+    "properties": {
+        "name": { "type": "string" },
+        "namespace": { "type": "string" },
+        "templates": { 
+            "type": "array",
+            "items": { "$ref": "#/definitions/HTMLTemplate"}
+        },
+        "js": {
+            "type": "array",
+            "items": { "$ref": "#/definitions/JSFile"}
+        },
+        "defaultComponentClass": {
+            "type": "array",
+            "items": { 
+                "$ref": "#/definitions/Class"
+            }
+        },
+        "css": {
+            "$ref": "#/definitions/CSSFile"
+        }
+    },
+    "definitions": {
+        "HTMLTemplate": {
+            "type": "object",
+            "properties": {
+                "fileType": { "$ref": "#/definitions/FileType" },
+                "fileName": { "type": "string" },
+                "children": {
+                    "type": "array",
+                    "items": { 
+                        "$ref": "#/definitions/CustomElementReference"
+                    }
+                },
+                "staticResources": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/StaticResourceReference"
+                    }
+                },
+                "dynamicChildren": {
```

Comment:
Can you provide more details on what `dynamicChildren` represents in comparison with `children`?

Reply:
It has a 'lwc:dynamic' directive. Updated the definition of DynamicComponentReference to add on to CustomElementReference. Will add a description to each property.